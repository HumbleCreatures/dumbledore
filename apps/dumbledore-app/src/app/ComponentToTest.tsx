
export const ComponentToTest = ({message, clickAction}: {message?:string, clickAction?: (message:string) => void}) => {
    if(!message) {
        return <div><div>This is not the message you are looking for.</div></div>
    }
    return <div>
        Message:{message}
        <div><button onClick={() => clickAction && clickAction(message)}>Klicka pÃ¥ mig</button></div>
        </div>
}