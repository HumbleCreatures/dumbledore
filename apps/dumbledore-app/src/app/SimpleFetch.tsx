import { QueryClient, useQuery, QueryClientProvider } from 'react-query';
type Case = {
    id: string;
    title: string;
    description: string;
    closed: boolean;
    created: Date;
}

export const SimpleFetchView = () => {
    const  { isLoading, error, data }   = useQuery('repoData', () =>
        fetch('http://localhost:3333/api/case/all').then(res =>
            res.json()
        )
    )

    if(isLoading) {
        return <div>Laddar data</div>
    }

    if(error) {
        return <div>Det gick inte sÃ¥ bra</div>
    }

    const listOfTesters = data?.map((item:Case) => { return <li>{item.title}</li>})

    return <ul>{listOfTesters}</ul>

}

const queryClient = new QueryClient()
export const SimpleFetchWithProviders = () => {
    return <QueryClientProvider client={queryClient}>
        <SimpleFetchView />
    </QueryClientProvider>
}