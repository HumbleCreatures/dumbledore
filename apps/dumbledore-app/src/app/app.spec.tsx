import { findByText, fireEvent, render } from '@testing-library/react';

import App from './app';
import { ComponentToTest } from './ComponentToTest';
import { SimpleFetchWithProviders } from './SimpleFetch';
import nock from 'nock';

describe('App', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<App />);

    expect(baseElement).toBeTruthy();
  });

  it('should have a greeting as the title', () => {
    const { getByText } = render(<ComponentToTest />);
    
    expect(getByText(/This is not the message you are looking for/gi)).toBeTruthy();
  });

  it('should have a greeting as the title', () => {
    const { getByText } = render(<ComponentToTest message="MEEP MEEP" />);
    
    expect(getByText(/MEEP MEEP/gi)).toBeTruthy();
  });

  it('snappshot test', () => {
    const { getByText } = render(<ComponentToTest message="MEEP MEEP" />);
    
    expect(getByText(/MEEP MEEP/gi)).toMatchSnapshot();
  });

  it('event function test', () => {
    const testFunction = jest.fn();
    
    const { getByText } = render(<ComponentToTest 
      message="MEEP MEEP" 
      clickAction={testFunction}
      />);
    const button = getByText('Klicka p√• mig');
    
    fireEvent(
      button,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    )

    expect(testFunction).toBeCalledTimes(1);
    expect(testFunction).toBeCalledWith("MEEP MEEP");
  });

});


/*

fireEvent(
  getByText(container, 'Submit'),
  new MouseEvent('click', {
    bubbles: true,
    cancelable: true,
  }),
)

fireEvent.change(input, {target: {value: '2020-05-24'}})
*/

/*

const nock = require('nock')

const scope = nock('https://api.github.com')
  .get('/repos/atom/atom/license')
  .reply(200, {
    license: {
      key: 'mit',
      name: 'MIT License',
      spdx_id: 'MIT',
      url: 'https://api.github.com/licenses/mit',
      node_id: 'MDc6TGljZW5zZTEz',
    },
  })
  
*/