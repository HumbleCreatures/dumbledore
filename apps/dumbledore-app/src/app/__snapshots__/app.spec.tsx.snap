// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snappshot test 1`] = `
<div>
  Message:
  MEEP MEEP
  <div>
    <button>
      Klicka p√• mig
    </button>
  </div>
</div>
`;
